bundles/alpha.model/src/alpha/model/util/Facet.xtend

```diff
@@ -77,18 +86,82 @@ class Facet {
 	
 	/** Extract the information from the given set. */
 	new(ISLBasicSet basicSet, ArrayList<Integer> saturatedInequalityIndices) {
-		equalities = DomainOperations.toISLEqualityMatrix(basicSet)
-		indexCount = basicSet.dim(ISLDimType.isl_dim_set)
-		indexInequalities = getInequalities(basicSet, indexCount, true)
+		val basicSetNoRedundancies = basicSet.copy.removeRedundancies
+		
+		space = basicSetNoRedundancies.space 
+		equalities = DomainOperations.toISLEqualityMatrix(basicSetNoRedundancies)
+		indexCount = basicSetNoRedundancies.dim(ISLDimType.isl_dim_set)
+		
+		val allInequalities = getInequalities(basicSetNoRedundancies, indexCount, true)
+		val inequalities = separateEffectivelySaturatedInequalities(allInequalities, space)
```

Comment:
Key difference is here: now we explicitly make a distinction between inequality constraints that are true ("effectively unsaturated”) and thick equalities (“effectively saturated”). The construction of the lattice subsequently only explores combinations of saturating the true inequality constraints.